src/note.rs

```diff
@@ -0,0 +1,48 @@
+macro_rules! parse_env {
+	($name:expr $(,)?) => {
+		::const_parse::parse_u128(::core::env!($name)) as u32
+	};
+}
+
+#[macro_export]
+macro_rules! define_abi_tag {
+	() => {
+		#[used]
+		#[unsafe(link_section = ".note.ABI-tag")]
+		static ABI_TAG: $crate::note::_AbiTag = $crate::note::_AbiTag::new();
+	};
+}
+
+#[repr(C)]
+#[doc(hidden)]
+pub struct _AbiTag {
```

Comment:
It's fine as is, but wouldn't something like `_hermit_version` or `_hermit_version_tag` be a little easier to comprehend?